cpp/src/arrow/ipc/metadata_internal.cc

```diff
@@ -988,11 +988,25 @@ Status MakeSparseTensorIndexCOO(FBB& fbb, const SparseCOOIndex& sparse_index,
   return Status::OK();
 }
 
-Status MakeSparseMatrixIndexCSR(FBB& fbb, const SparseCSRIndex& sparse_index,
+template <typename SparseIndexType>
```

Comment:
Perhaps we could put those helpers in the anonymous namespace?

Reply:
Yes, I should put it in the anonymous namespace.  I will do it.